src/org.xtuml.bp.ui.text/src/org/xtuml/bp/ui/text/editor/oal/OALAutoEditStrategy.java

```diff
@@ -0,0 +1,88 @@
+//========================================================================
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not 
+// use this file except in compliance with the License.  You may obtain a copy 
+// of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT 
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   See the 
+// License for the specific language governing permissions and limitations under
+// the License.
+//========================================================================
+
+package org.xtuml.bp.ui.text.editor.oal;
+
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.DocumentCommand;
+import org.eclipse.jface.text.IAutoEditStrategy;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.ui.editors.text.EditorsUI;
+import org.eclipse.ui.texteditor.AbstractDecoratedTextEditorPreferenceConstants;
+import org.xtuml.bp.core.CorePlugin;
+
+public class OALAutoEditStrategy implements IAutoEditStrategy {
+
+    public void customizeDocumentCommand(IDocument document, DocumentCommand command) {
+        try {
+            if (command.text.equals(System.lineSeparator())) {
+                int line = document.getLineOfOffset(command.offset);
+                String curIndent = getIndentOfLine(document, line);
+                String nextIndent = getAdditionalIndent(document, line);
+                command.text = System.lineSeparator() + curIndent + nextIndent;
+            }
+        } catch (BadLocationException ble) {
+            CorePlugin.logError("Invalid edit location exception caught.", ble);
+        }
+    }
+
+    public static int findEndOfWhiteSpace(IDocument document, int offset, int end) throws BadLocationException {
+        while (offset < end) {
+            char c = document.getChar(offset);
+            if (c != ' ' & c != '\t') {

```

Comment:
This leaves out CR/LF. Perhaps there is a better way to do this. There is a function available to check for whitespace.